e2e/kms.go

```diff
@@ -107,3 +119,68 @@ func (vc *vaultConfig) getPassphrase(f *framework.Framework, key string) (string
 
 	return strings.TrimSpace(stdOut), strings.TrimSpace(stdErr)
 }
+
+// creatEncSetup sets up required resources for encryption with given kmsID
+func createEncSetup(f *framework.Framework, kmsID string) {
+	switch kmsID {
+	case "vault-tokens-test":
+		// name(space) of the Tenant
+		tenant := f.UniqueName
+
+		// create the Secret with Vault Token in the Tenants namespace
+		createSecret(f.ClientSet, vaultExamplePath+"tenant-token.yaml", tenant)
+
+	case "vault-tenant-sa-test":
+		err := createTenantServiceAccount(f.ClientSet, f.UniqueName)
+		if err != nil {
+			e2elog.Failf("failed to create ServiceAccount: %v", err)
+		}
+	case "user-ns-secrets-metadata-test":
+		// user provided namespace where secret will be created
+		namespace := cephCSINamespace
+
+		// create user Secret
+		createSecret(f.ClientSet, vaultExamplePath+"user-secret.yaml", namespace)
+	case "user-secrets-metadata-test":
+		// PVC creation namespace where secret will be created
+		namespace := f.UniqueName
+
+		// create user Secret
+		createSecret(f.ClientSet, vaultExamplePath+"user-secret.yaml", namespace)
+
+	default:
+		// no actions required for other encryption kmsID
+	}
+}
+
+// deleteEncSetup tears down up required resources for encryption with given kmsID
+func deleteEncSetup(f *framework.Framework, kmsID string) {
```

Comment:
deleteEncSetup and createEncSetup look similar. pass KMsID and operation type to avoid code duplication?